VideoLocker/src/test/java/org/edx/mobile/view/CourseUnitNavigationActivityTest.java

```diff
@@ -438,12 +441,16 @@ public void testUnitFragmentCreation() {
         when(othersModel.getType()).thenReturn(BlockType.OTHERS);
         unitList.add(othersModel);
 
-        CourseUnitPagerAdapter adapter = new CourseUnitPagerAdapter(fragmentManager, unitList,
-                courseData, hasComponent);
+        CourseUnitPagerAdapter adapter = new CourseUnitPagerAdapter(fragmentManager, config,
+                unitList, courseData, hasComponent);
 
         assertTrue(adapter.getItem(0) instanceof CourseUnitVideoFragment);
         assertTrue(adapter.getItem(1) instanceof CourseUnitOnlyOnYoutubeFragment);
-        assertTrue(adapter.getItem(2) instanceof CourseUnitDiscussionFragment);
+        if (config.isDiscussionsEnabled()) {
+            assertTrue(adapter.getItem(2) instanceof CourseUnitDiscussionFragment);
+        } else {
+            assertTrue(adapter.getItem(2) instanceof CourseUnitMobileNotSupportedFragment);
+        }
         assertTrue(adapter.getItem(3) instanceof CourseUnitMobileNotSupportedFragment);
         assertTrue(adapter.getItem(4) instanceof CourseUnitWebViewFragment);
         assertTrue(adapter.getItem(5) instanceof CourseUnitEmptyFragment);
```

Comment:
I think using indices is not the best coding pattern, since it's too vulnerable to typos/ordering errors. It might be better to collect these expected classes in a list that's populated along with the provided models, then iterate over it and assert that the adapter item at those indices match those of the classes you expect.